From 3afc67bb55fbdb3259397b3199bf9efef84bb129 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sun, 2 Aug 2015 20:16:09 -0700
Subject: [PATCH 009/192] Snap: Re-enable ZSL after exiting HDR mode

Author: Sultanxda <sultanxda@gmail.com>
Date:   Sun Aug 2 20:16:09 2015 -0700

    SnapdragonCamera: Re-enable ZSL after exiting HDR mode

    * Make sure to check that HDR was changed before forcing ZSL back on.

    Change-Id: Ifb76263c3a50b919762fd2a313d6013295c0619f

Author: Gabriele M <moto.falcon.git@gmail.com>
Date:   Tue Jan 19 16:45:25 2016 +0100

    Snap: Don't enable ZSL when disabling HDR

    Instead, restore the default value. This prevents issues on devices
    that don't support ZSL.

    Change-Id: I9c456cd6503df29800e2bc7ccaa691cb3fb42681

Change-Id: I66c9e31f37ef09c2437fcacac6c04246dab7df3c
---
 src/com/android/camera/PhotoModule.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index cd78384f1..6fd9a2636 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4772,6 +4772,8 @@ public class PhotoModule
 
         //filter off unsupported settings
         final String settingOff = mActivity.getString(R.string.setting_off_value);
+        final String settingOn = mActivity.getString(R.string.setting_on_value);
+        final String zsl = mActivity.getString(R.string.pref_camera_zsl_default);
         if (!CameraSettings.isZSLHDRSupported(mParameters)) {
             //HDR internally uses AE-bracketing. Disable both if not supported.
             if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOff) ||
@@ -4780,6 +4782,9 @@ public class PhotoModule
             } else if (notSame(pref,CameraSettings.KEY_ZSL,settingOff)) {
                 mUI.setPreference(CameraSettings.KEY_CAMERA_HDR, settingOff);
                 mUI.setPreference(CameraSettings.KEY_AE_BRACKET_HDR, settingOff);
+            } else if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOn) ||
+                notSame(pref, CameraSettings.KEY_AE_BRACKET_HDR, settingOn)) {
+                mUI.setPreference(CameraSettings.KEY_ZSL, zsl);
             }
         }
 
-- 
2.17.1

